### PR TITLE
Fix support of cluster_backend=docker

### PR DESCRIPTION
**Pull request by fruch**
trello task: https://trello.com/c/utlFCXLp

PLEASE DON'T SQUASH THIS PR.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- ~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~
- ~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1121,7 +1136,7 @@ def _report_housekeeping_uuid(self, verbose=False):
         if uuid_result.exit_status == 0 and mark_result.exit_status != 0:
             result = self.remoter.run('cat %s' % uuid_path, verbose=verbose)
             self.remoter.run(cmd % result.stdout.strip(), ignore_status=True)
-            self.remoter.run('sudo -u scylla touch %s' % mark_path,
```
Comment (fruch) [inline sdcm/cluster.py]: recheck if still needed

File: sdcm/docker.py [hunk 2]
```diff
@@ -102,6 +96,32 @@ def destroy(self, force=True):  # pylint: disable=arguments-differ
         force_param = '-f' if force else ''
         _cmd('rm {} -v {}'.format(force_param, self.name))
 
+    def start_scylla_server(self, verify_up=True, verify_down=False, timeout=300):
```
Comment (fruch) [inline sdcm/docker.py]: check again if we need to re-implement this

File: sdcm/docker.py [hunk 3]
```diff
@@ -215,33 +239,150 @@ def destroy(self):
...
+                                                  cluster_prefix=cluster_prefix,
                                                   **kwargs)
 
-    @retrying(n=30, sleep_time=3, allowed_exceptions=(cluster.ClusterNodesNotReady, DockerCommandError))
+    def node_setup(self, node, verbose=False, timeout=3600):
+        endpoint_snitch = self.params.get('endpoint_snitch')
+        seed_address = ','.join(self.seed_nodes_ips)
+
+        node.wait_ssh_up(verbose=verbose)
+        # node.install_scylla_debuginfo()
```
Comment (fruch) [inline sdcm/docker.py]: think if we want this every time, take a while to download and install

File: sdcm/docker.py [hunk 4]
```diff
@@ -215,33 +239,150 @@ def destroy(self):
...
+
+    @cluster.wait_for_init_wrap
     def wait_for_init(self, node_list=None, verbose=False, timeout=None):   # pylint: disable=unused-argument,arguments-differ
         node_list = node_list if node_list else self.nodes
         for node in node_list:
             node.wait_for_status_running()
-        return self.check_nodes_up_and_normal(node_list)
+        self.wait_for_nodes_up_and_normal(node_list)
+
+    def get_scylla_args(self):
```
Comment (fruch) [inline sdcm/docker.py]: thinks if to remove much more here, since crashing with the default bad alloc flags

File: sdcm/docker.py (hunk 3 above)
Comment (fruch) [reply sdcm/docker.py]: also need to make sure it takes the correct ones

File: sdcm/docker.py (hunk 2 above)
Comment (fruch) [reply sdcm/docker.py]: seem like the current scylla-server.service doesn't work out of the box, I'll leave it to fallback to supervisor

File: sdcm/docker.py (hunk 3 above)
Comment (fruch) [reply sdcm/docker.py]: maybe we should put a configuration parameter to enable disable the install of debug info ? so the default would be speedy

Comment (fruch) [reply sdcm/docker.py]: it's now will be under `if Setup.BACKTRACE_DECODING:`

File: sdcm/docker.py (hunk 4 above)
Comment (fruch) [reply sdcm/docker.py]: check again, seem like now need

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py]: fixed this

File: sdcm/docker.py [hunk 5]
```diff
@@ -215,33 +240,151 @@ def destroy(self):
...
+        if self.params.get('client_encrypt'):
+            node.config_client_encrypt()
+
+
+def send_receive_files(self, src, dst, delete_dst=False, preserve_perm=True, preserve_symlinks=False):  # pylint: disable=too-many-arguments,unused-argument
+    if src != dst:
+        self.remoter.run(f'cp {src} {dst}')
+
+
+class FakeNode(cluster.BaseNode):  # pylint: disable=abstract-method,too-many-instance-attributes
```
Comment (bentsi) [inline sdcm/docker.py]: DummyMonitoringNode

File: README.rst [hunk 6]
```diff
@@ -195,6 +195,15 @@ With that said and done, you can run your test using the command line::
...
 
 
+Docker
+------
+
+We can also enable running with scylla formal docker images::
+
+    # example of running specific docker version
+    export SCT_SCYLLA_VERSION=3.2.rc1
+    hydra run-test longevity_test.LongevityTest.test_custom_time --backend docker --config test-cases/PR-provision-test-docker.yaml
```
Comment (fruch) [inline README.rst]: add docs for sudo and passwordless access needed

File: docker/env/version [hunk 7]
```diff
@@ -1 +1 @@
-0.57
+0.58-fix-docker-2
```
Comment (fruch) [inline docker/env/version]: change name before merge

**Comment by fruch** [comment]
@bentsi address the issues, a reminder, please `rebase and merge`

File: test-cases/PR-provision-test-docker.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+test_duration: 15
```
Comment (bentsi) [inline test-cases/PR-provision-test-docker.yaml]: why do we have special YAML file for Docker?

Comment (fruch) [reply test-cases/PR-provision-test-docker.yaml]: I've started like that cause of `use_mgmt: true` that wasn't test fully (and take a longer time to run and install)
and scylla-bench isn't install to the docker loader version.

I can do both in a different PR and remove this file afterwards

**Comment by bentsi** [comment]
image
Please fix pre-commit issues

**Pull request merged**